### ISSUE TITLE
Why choose num_classes=1008 when creating the model?

### ISSUE THREAD

**Issue by jvhoffbauer** [mseitzer/pytorch-fid#43]
Hello, 

Is there a special reason as to why you choose `num_classes=1008` instead of using the default value `1000 ` when creating the model. Refer to: https://github.com/mseitzer/pytorch-fid/blob/master/pytorch_fid/inception.py#L193

Best, Jan

**Comment by jvhoffbauer** [comment]
I think it is related to this issue: https://github.com/tensorflow/tensorflow/issues/4128#issuecomment-244142481

**Comment by mseitzer** [comment]
Indeed, the version of the Inception model FID is using has 1008 classes. This does not really matter for FID, as the classification layer is not used in any way.

**Comment by jvhoffbauer** [comment]
Yes, thank you! I was looking into implementing Inception Score and using your InceptionV3 implementation for other tasks. 

Would you agree, that one can just truncate the output of the last layer to the first 1000 classes when implementing the Inception Distance?

**Comment by FyreMael** [comment]
> (1) It turns out that the 1008 size softmax output is an artifact of dimension back-compatibility with a older, Google-internal system. Newer versions of the inception model have 1001 output classes, where one is an "other" class used in training. You shouldn't need to pay any attention to the extra 8 outputs.

https://github.com/tensorflow/tensorflow/issues/4128#issuecomment-244142481

**Comment by mseitzer** [comment]
The original Inception score (https://github.com/openai/improved-gan/) uses the same network (weights) as we use here. On top of that, apparently the original Inception score *uses all 1008* classes to compute the score: https://github.com/openai/improved-gan/issues/29. This is a bug.

However, if you want to stay compatible to Inception scores reported earlier, you also need to take this bug into account in your implementation. If you just want to compare different models where you compute the score for all of them yourself, you could consider implementing a slightly improved version of Inception score: https://arxiv.org/pdf/1801.01973.pdf.
